src/lcdfonts.jl

```diff
@@ -0,0 +1,256 @@
+"ϕ" => "00000101101010110101011100010000100"*lcd_short,
+"ω" => "00000000000101010001101011010101010"*lcd_short,
+"Π" => "11111010100101001010010100101001010"*lcd_short,
+"χ" => "00000000001100101010001000101010011"*lcd_short,
+"Δ" => "00000001000101001010100011000111111"*lcd_short,
+"Δ" => "00000001000101001010100011000111111"*lcd_short,
+"κ" => "00000100010101001100011000101010001"*lcd_short,
+"□" => "00000111111000110001100011111100000"*lcd_short,
+"ν" => "00000000001000110010101000110001000"*lcd_short,
+"η" => "00000000001011011001100011000110001"*"000010000100001",
+"░" => "10101010101010101010101010101010101"*"010101010101010",
+"█" => lcd_filled,
+"λ" => "11000001000010000100001000101010001"*lcd_short,
+"τ" => "00000000001111100100001000010000011"*lcd_short,
+"ψ" => "00000001001010110101011100010000100"*lcd_short,
+"Ψ" => "10101101011010101110001000010001110"*lcd_short,
+"Λ" => "00100001000101001010010101000111011"*lcd_short,
+"Θ" => "01110100011000110101100011000101110"*lcd_short,
+"Φ" => "11111001000111010101011100010011111"*lcd_short,
+"†" => "00000001000111000100001000010000000"*lcd_short,
+"∠" => "00000000000000100010001000100011111"*lcd_short,
+"⟂" => "00000000000010000100001000010011111"*lcd_short,
+"≡" => "00000111110000011111000001111100000"*lcd_short,
+"±" => "00100001001111100100001000000011111"*lcd_short,
+"∓" => "11111000000010000100111110010000100"*lcd_short,
+"∇" => "00000111111000110001010100101000100"*lcd_short,
+"∂" => "00110000010000101101100011000101110"*lcd_short,
+"≠" => "00001000101111100100111110100010000"*lcd_short,
+"𝚤" => "00100000000110000100001000010100110"*lcd_short,
+" " => lcd_blank
+)
+
+const scripted_equation = "H=ħωσ_x+Jσ_y\n\nc^2=a^2+b^2\n\n|Ψ>=a^†a|ψ_1>+b^†b|ψ_2>"
+const test_string = string(
+"Keyboard characters:
+
+ABCDEFGHIJKLMNOPQRSTUVWXYZ
+abcdefghijklmnopqrstuvwxyz
+
+~!@#\$%^&*()\_+
+`1234567890-=
+
+[]\\{}|;':\",./<>?
+
+Non-keyboard characters:
+
+αβγδϵηθκλμνπρστϕχψωħ
+ΩΣΞΓΠΨΔΛΘΦ
+𝚤∞÷√±∓≠≡⟂∠°∂∇†□░█\n\n",
+length(lcd), " characters supported so far.")
+
+function lcdstring!(string_cell::Cell, str::String, pixelsize, pixelspacing; meta::Meta=GDSMeta(0,0), scripting = false, linelimit = 2^32)
+    pushstr = "push!(string_cell.refs, CellReference(s_cell, Point(pixelspacing*6*(hpos - 1), -11*pixelspacing*(vpos - offset))))"
+    hpos = 1
+    vpos = 1
+    subscript = -1
+    superscript = +1
+    for s in str
+        offset = 0.0
+        if subscript == 0
+            offset = -0.3
+            subscript = -1
+        elseif superscript == 0
+            offset = +0.3
+            superscript = +1
+        end
+        if s == '\n' || hpos > linelimit
+            vpos += 1
+            hpos = 1
+        elseif s == '_' && scripting
+            subscript = +1
+        elseif s == '^' && scripting
+            superscript = -1
+        else
+            s = string(s)
+            ss = get(lcd, s, s)
+            try
+                s_cell = drawlcdcell(ss, pixelsize, pixelspacing, meta)
```

Comment:
One last note, which you don't have to address right now. It looks like you're making a CellReference for each letter, but not reusing the referenced cell (in other words, there could be several cells that contain nothing but the same character). If you don't reuse the cells, you could have just drawn directly rather than putting each character in its own cell. This would avoid cluttering the GDS file with N cells for text of length N. An alternative would be to save a cache of character cells that you've used, so that there's at most 26 cells for text of length N (provided you used only letters). But that would be a lot more work for not much gain unless you have a huge amount of text in your GDS file.